sambacc/typelets.py

```diff
@@ -20,8 +20,17 @@
 """
 
 from types import TracebackType
+import sys
 import typing
 
 ExcType = typing.Optional[typing.Type[BaseException]]
 ExcValue = typing.Optional[BaseException]
 ExcTraceback = typing.Optional[TracebackType]
+
+
+if sys.version_info >= (3, 11):
+    from typing import Self
+elif typing.TYPE_CHECKING:
+    from typing_extensions import Self
+else:
+    Self = typing.Any
```

Comment:
Nice idea!

When checking python version, I usually use [platform.python_version_tuple](https://docs.python.org/3/library/platform.html#platform.python_version_tuple), but `sys.version_info` is just fine here.